src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -316,5 +335,71 @@ public async Task<bool> KeepAliveJobAsync(JobInfo jobInfo, CancellationToken can
                 throw;
             }
         }
+
+        public async Task ArchiveJobsAsync(byte queueType, CancellationToken cancellationToken)
+        {
+            using SqlConnectionWrapper conn = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, false);
+            using SqlCommandWrapper cmd = conn.CreateRetrySqlCommand();
+            VLatest.ArchiveJobs.PopulateCommand(cmd, queueType);
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        public async Task ExecuteWithHeartbeatAsync(byte queueType, long jobId, long version, Func<CancellationToken, Task> action, TimeSpan heartbeatPeriod, CancellationToken cancellationToken)
```

Comment:
Should this be in queueclient?

Reply:
Yes, as it deals with heartbeat on jobs in the queue.